File: Programming Language(ver.2)/Go/608.go
package main
import (
	"fmt"
)

func min(x, y int) int {
	if x < y {
		return x
	}else{
		return y
	}
}

func main(){
	var array = make([]int, 5)
	var score = 0;
	for i := 0; i < 10; i++ {
		var temp int
		fmt.Scan(&temp)
		if temp == 0 {
			continue
		}
		array[0] = 1
		for j := 3; j >= 0; j-- {
			array[min(j+temp, 4)] += array[j]
			array[j] = 0
		}
		if array[4] > 0{
			score += array[4]
			array[4] = 0
		}
	}
	fmt.Printf("score = %d\n", score)
}

File: Programming Language(ver.2)/PHP/608.php
<?php
    $array = array();
    for($i = 0; $i < 5; $i++){
        array_push($array, 0);
    }
    $score = 0;
    for($i = 0; $i < 10; $i++){
        $task = intval(fgets(STDIN));
        if($task == 0){
            continue;
        }
        $array[0] = 1;
        for($j = 3; $j >= 0; $j--){
            $array[min($j+$task, 4)] += $array[$j];
            $array[$j] = 0;
        }
        if($array[4] > 0){
            $score += $array[4];
            $array[4] = 0;
        }
    }
    printf("score = %d\n", $score);
?>